device.go

```diff
@@ -142,6 +149,20 @@ func (d *Device) Config(cfgNum int) (*Config, error) {
 	return cfg, nil
 }
 
+func (d *Device) detachKernelDriver(cfg *Config) (err error) {
```

Comment:
I think it might be slightly better to inline the body of this function directly inside Config(), as it's not used anywhere else.
Then it becomes simply:

if d.autodetach {
  for _, iface := range cfg.Desc.Interfaces {
    if err := ....; err != nil {
      return nil, fmt.Errorf("Can't detach kernel driver ...")
    }
  }
}